### PR TITLE
[homekit] Update for Xcode 10 beta 2

### PR DESCRIPTION
**Pull request by VincentDondain**

### CODE REVIEW

File: src/homekit.cs [hunk 1]
```diff
@@ -714,14 +714,14 @@ partial interface HMHome {
...
-		[Availability (Introduced = Platform.iOS_8_0, Deprecated = Platform.iOS_9_0)]
+		[Availability (Introduced = Platform.iOS_8_0, Deprecated = Platform.iOS_9_0, Message = "Use 'ManageUsers' instead.")]
 		[Async]
 		[Export ("addUserWithCompletionHandler:")]
 		void AddUser (Action<HMUser,NSError> completion);
 
 		[NoTV]
 		[NoWatch]
-		[Availability (Introduced = Platform.iOS_8_0, Deprecated = Platform.iOS_9_0)]
+		[Availability (Introduced = Platform.iOS_8_0, Deprecated = Platform.iOS_9_0, Message = "Use 'ManageUsers' instead.")]
```
Comment (spouliot) [inline src/homekit.cs]: I did not clean the beta 2 diff (like the beta 1) since there's a _typo?_ there. Did you spot it (and file a rdar) ?

Comment (VincentDondain) [reply src/homekit.cs]: I'm sorry I don't understand what you're trying to say (: What typo? What rdar?

Comment (VincentDondain) [reply src/homekit.cs]: The diff for beta 2 is just the addition of a message with the better API to use.

Comment (spouliot) [reply src/homekit.cs]: no, read it again :)

Comment (VincentDondain) [reply src/homekit.cs]:
``` diff
-- (void)addUserWithCompletionHandler:(void (^)(HMUser * __nullable user, NSError * __nullable error))completion API_PROHIBITED(watchos, tvos) API_DEPRECATED("No longer supported.", ios(8.0, 9.0));
+- (void)addUserWithCompletionHandler:(void (^)(HMUser * __nullable user, NSError * __nullable error))completion API_UNAVAILABLE(macos, watchos, tvos) API_DEPRECATED_WITH_REPLACEMENT("-manageUsersWithCompletionHandler:", ios(8.0, 9.0));
 ```

and

``` diff
-- (void)removeUser:(HMUser *)user completionHandler:(void (^)(NSError * __nullable error))completion API_PROHIBITED(watchos, tvos) API_DEPRECATED("No longer supported.", ios(8.0, 9.0));
+- (void)removeUser:(HMUser *)user completionHandler:(void (^)(NSError * __nullable error))completion API_PROHIBITED(macos, ios, watchos, tvos) API_DEPRECATED_WITH_REPLACEMENT("-manageUsersWithCompletionHandler:", ios(8.0, 9.0));
```

What typo do you see I only see a move from `API_PROHIBITED` to `API_UNAVAILABLE` and `API_DEPRECATED` to `API_DEPRECATED_WITH_REPLACEMENT`

Comment (dalexsoto) [reply src/homekit.cs]: @VincentDondain I think @spouliot  means

```diff
-- API_PROHIBITED(watchos, tvos) API_DEPRECATED("No longer supported.", ios(8.0, 9.0));
+- API_UNAVAILABLE(macos, watchos, tvos) API_DEPRECATED_WITH_REPLACEMENT("-manageUsersWithCompletionHandler:", ios(8.0, 9.0));
```

more specifically 

```diff
-- API_PROHIBITED(watchos, tvos)
+- API_UNAVAILABLE(macos, watchos, tvos)
```

Comment (VincentDondain) [reply src/homekit.cs]: But HomeKit is not available yet on macOS...

Comment (dalexsoto) [reply src/homekit.cs]: giphy

Comment (VincentDondain) [reply src/homekit.cs]: Shame

Comment (spouliot) [reply src/homekit.cs]: @VincentDondain it's weird (and not symmetrical to the `add` API) so it's likely a mistake. Add it to the xcode10 tracking issue and file a rdar :) thanks!

**Pull request merged**